Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -1216,23 +1216,35 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
     legacyMainFrameProcess().send(Messages::WebPage::EnableTextAnimationTypeForElementWithID(elementID, uuid), webPageIDInMainFrameProcess());
 }
 
-void WebPageProxy::addTextAnimationForAnimationID(IPC::Connection& connection, const WTF::UUID& uuid, const TextAnimationData& styleData, const WebCore::TextIndicatorData& indicatorData)
+void WebPageProxy::addTextAnimationForAnimationID(IPC::Connection& connection, const WTF::UUID& uuid, const TextAnimationData& styleData, const WebCore::TextIndicatorData& indicatorData, CompletionHandler<void()>&& completionHandler)
 {
     MESSAGE_CHECK(uuid.isValid());
 
-    internals().textIndicatorDataForChunk.add(uuid, indicatorData);
+    internals().textIndicatorDataForAnimationID.add(uuid, indicatorData);
+
+    if (completionHandler)
+        internals().completionHandlerForAnimationID.add(uuid, WTFMove(completionHandler));
 
     protectedPageClient()->addTextAnimationForAnimationID(uuid, styleData);
 }
 
+void WebPageProxy::callCompletionHandlerForAnimationID(const WTF::UUID& uuid)
```

Comment:
If the web process dies, I think `completionHandlerForAnimationID` is going to be left with stale completion handlers.

`WebPageProxy::resetState` should probably clear the map.